.github/workflows/tests.yml

```diff
@@ -1,28 +1,57 @@
 name: Run Tests
 
-on: [push, pull_request, workflow_dispatch]
+on: 
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
 
 jobs:
   unittest-and-lint:
-    runs-on: ubuntu-latest
+
+    runs-on: ${{matrix.os}}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        python-version: ["3.8", "3.10"]
+
     steps:
-    - uses: actions/checkout@v1
-    - name: Install dependencies
-      run: |
-        python3 -m pip install --upgrade pip
-        python3 -m pip install -r requirements.txt
-        python3 -m pip install types-Deprecated coveralls mypy pytest pytest-cov
-    - name: Create somersault.pdx
-      run: |
-        export PYTHONPATH=$PYTHONPATH:$PWD
-        cd examples
-        python3 mksomersaultpdx.py somersault.pdx
-    - name: Run unit tests
-      run: |
-        export PYTHONPATH=$PYTHONPATH:$PWD
-        pytest --tb=native --cov=odxtools --cov-report=term --cov-report=lcov .
-    - name: Static type checking with mypy
-      run: |
-        python3 -m mypy --ignore-missing-imports --python-version 3.8 odxtools tests examples
-        python3 -m mypy --ignore-missing-imports --python-version 3.9 odxtools tests examples
-        python3 -m mypy --ignore-missing-imports --python-version 3.10 odxtools tests examples
+      - uses: actions/checkout@v3
+
+      - name: Set up Python ${{matrix.python-version}}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{matrix.python-version}}
+
+      - name: Install odxtools dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+      
+      # we need additional dependencies for testing and the type checker
+      # but we avoided to specify as hard dependencies within 'requirements.txt'
+      - name: Install test dependencies
+        run:  |
+          pip install types-Deprecated coveralls mypy pytest pytest-cov
+
+      - name: Create somersault.pdx
+        run: |
+          python -m examples.mksomersaultpdx examples/somersault.pdx
+
+      - name: Run unit tests and report line coverage
+        # use `python -m pytest ...` instead of `pytest ...` to avoid setting the
+        # PYTHONPATH environment variable. This works independently of the OS.
+
+        # For some reason, pytest-cov reports invalid line coverage. As a remedy,
+        # we utilize the `coverage` module instead of `pytest-cov`. Line coverage
```

Comment:
* then why is pytest-cov in the dependency list if not used?
* sure this issue happens in latest version? pytest-cov uses the coverage module internally

Reply:
Unfortunately, yes. `pytest-cov` reports only 36 % coverage in contrast to 65 % reported by `coverage`. For some reason, `pytest-cov` marks every statement on module level (i.e. imports, method signatures, constants) as uncovered, which seems wrong from my point of view.